### PR TITLE
boehmgc: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Description of changes

- Add myself as maintainer
- Add some options

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Not sure the branch change went right.

**Comment by AndersonTorres** [comment]
100% sure the branch change failed miserably. Fixing.

**Comment by bobby285271** [comment]
@ofborg build guile

**Review by bobby285271 — DISMISSED** [review]
guile still fails (cherry-picked to master and tested native build, x86_64-linux), note that this is [experimental release](https://github.com/ivmai/bdwgc/commit/47e9106c17b72e9ee5501308f69ea94531e798b3) and previously reverted in https://github.com/NixOS/nixpkgs/pull/152145#issuecomment-1001133645


<details>

```
Making all in libguile
make[2]: Entering directory '/build/guile-2.2.7/libguile'
  GEN      libpath.h
  SNARF  alist.x
In file included from ../libguile/bdw-gc.h:47,
                 from ../libguile/gc.h:156,
                 from ../libguile/pairs.h:28,
                 from ../libguile/_scm.h:74,
                 from alist.c:25:
/nix/store/iw749gq7s1szysr688llqc65d2k2r8a5-boehm-gc-8.2.0-dev/include/gc/gc.h:1784:11: fatal error: gc_pthread_redirects.h: No such file or directory
 1784 | # include "gc_pthread_redirects.h"
      |           ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:4264: alist.x] Error 1
make[2]: Leaving directory '/build/guile-2.2.7/libguile'
make[1]: *** [Makefile:1857: all-recursive] Error 1
make[1]: Leaving directory '/build/guile-2.2.7'
make: *** [Makefile:1743: all] Error 2
```

</details>

**Comment by AndersonTorres** [comment]
Being experimental, I will remove the "upgrade" commit.

**Comment by AndersonTorres** [comment]
@ofborg build guile

**Comment by bobby285271** [comment]
May target master if there is no rebuilds. I am not familiar with this pkg, ask for someone else if you need a review :-)

**Comment by AndersonTorres** [comment]
@ofborg build guile

**Comment by AndersonTorres** [comment]
@ofborg build guile

File: pkgs/development/libraries/boehm-gc/default.nix [hunk 1]
```diff
@@ -22,22 +27,22 @@ stdenv.mkDerivation rec {
...
+  ]
+  ++ lib.optional enableMmap "--enable-mmap"
+  ++ lib.optional enableLargeConfig "--enable-large-config";
 
-  nativeBuildInputs =
-    lib.optional stdenv.hostPlatform.isMinGW autoreconfHook;
+  nativeBuildInputs = lib.optional stdenv.hostPlatform.isMinGW autoreconfHook;
 
-  doCheck = true; # not cross;
+  doCheck = true; # not cross
```
Comment (Mindavi) [inline pkgs/development/libraries/boehm-gc/default.nix:39]: Nit: I think 'not cross' is implied, no need to keep it here

**Comment by Mindavi** [comment]
Diff LGTM, have you tried building and running nix with this?

**Comment by AndersonTorres** [comment]
@ofborg build guile nix

**Comment by AndersonTorres** [comment]
@ofborg build guile nix

File: pkgs/development/libraries/boehm-gc/default.nix [hunk 2]
```diff
@@ -54,13 +59,12 @@ stdenv.mkDerivation rec {
...
     license = "https://hboehm.info/gc/license.txt";
-
-    maintainers = [ ];
-    platforms = lib.platforms.all;
+    maintainers = with maintainers; [ AndersonTorres ];
+    platforms = platforms.all;
   };
-}
+
+  passthru.tests = nixVersions;
```
Comment (jonringer) [inline pkgs/development/libraries/boehm-gc/default.nix:69]: not sure how I feel about passthru being after meta. Personally, i don't expect there to be anything after meta, has this changed?

Comment (AndersonTorres) [reply pkgs/development/libraries/boehm-gc/default.nix:69]: To be honest I don't know. I found both ways in a cursory look.
I believe that both are a bit "meta" in the sense they do not trigger rebuilds.
I will revert this.

**Review by Mindavi — CHANGES_REQUESTED** [review]
One comment, otherwise looks ok. Can be retargeted to master again as long as nothing is changed that causes new rebuilds :)

File: pkgs/development/libraries/boehm-gc/default.nix [hunk 3]
```diff
@@ -1,19 +1,24 @@
-{ lib, stdenv, fetchurl
+{ config
+, lib
+, stdenv
+, fetchurl
 , autoreconfHook
-, enableLargeConfig ? false # doc: https://github.com/ivmai/bdwgc/blob/v8.0.6/doc/README.macros (LARGE_CONFIG)
+# doc: https://github.com/ivmai/bdwgc/blob/v8.0.6/doc/README.macros (LARGE_CONFIG)
+, enableLargeConfig ? config.boehmgc.enableLargeConfig or false
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/development/libraries/boehm-gc/default.nix:7]: The idea is to "override" the default value. Something wrong with this?

**Review by Mindavi — APPROVED** [review]

**Comment by bobby285271** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by AndersonTorres** [comment]
Any noticeable reason for the egregious failure on Darwintel?

**Comment by AndersonTorres** [comment]
@ofborg build guile nix

**Pull request merged**